checker.go

```diff
@@ -364,6 +364,50 @@ func (c *hasLenChecker) Check(got interface{}, args []interface{}, note func(key
 	return nil
 }
 
+// Implements checks that the provided value implements the given interface. A
+// pointer to the interface must be provided.
+//
+// For instance:
+//
+//     c.Assert(myReader, qt.Implements, io.ReadCloser)
+//
+var Implements Checker = &implementsChecker{
+	argNames: []string{"got", "want"},
+}
+
+type implementsChecker struct {
+	argNames
+}
+
+// Check implements Checker.Check by checking that got implements the
+// interface pointed to by args[0].
+func (c *implementsChecker) Check(got interface{}, args []interface{}, note func(key string, value interface{})) (err error) {
+	if got == nil {
+		note("error", Unquoted("got nil value but want non-nil"))
+		note("got", got)
+		return ErrSilent
+	}
+
+	if args[0] == nil {
+		return BadCheckf("wanted value is nil but must be non-nil")
+	}
+	wantType := reflect.TypeOf(args[0])
+	if wantType.Kind() != reflect.Ptr || wantType.Elem().Kind() != reflect.Interface {
```

Comment:
This looks good. We could be more user friendly with the error message?
```go
if wantType.Kind() != reflect.Ptr {
	note("variable", args[0])
	return BadCheckf("want a pointer to an interface value but a non-pointer variable was provided")
}
if wantType.Elem().Kind() != reflect.Interface {
	note("pointed type", wantType.Elem().String())
	return BadCheckf("want a pointer to an interface value but a pointer to a concrete type was provided")
}
```
Or something similar? Take it or leave it.